duckduckgo/macos-browser
# GPC Implementation

Task/Issue URL: https://app.asana.com/0/72649045549333/1200332991702960/f
Tech Design URL: https://app.asana.com/0/481882893211075/1201196518499677/f
CC: @tomasstrba @samsymons 

**Description**:
This PR implements the GPC spec into the browser. The PR adds a UserScript to add the GPC DOM property as well as a request header for pre-defined sites. Additionally a setting has been added in preferences to allow users to toggle GPC. This defaults to enabled.

**Note:** ~Different from the iOS implementation I have moved the pre-defined list of sites that we're sending the GPC header to into the privacy remote configuration. I am opening a PR to deploy this new version of the config. If during testing the header test fails on the test pages please comment out lines `64-67` and line `70` in `PrivacyConfigurationManager.swift`. This will force the browser to use the embedded config which has this list of sites already.~ This change should be deployed.

**Steps to test this PR**:
1. Visit https://global-privacy-control.glitch.me/
1. Server side detection and client side detection should have blue thumbs up
2. Visit https://privacy-test-pages.glitch.me/privacy-protections/gpc/
3. Click "start test"
4. Expand the test results
5. top frame header, top frame JS API, and frame JS API should display `1` or `true`
6. Open preferences and disable the GPC setting
7. Repeat the tests in steps 1-6
8. All tests should fail

**Testing checklist**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
